packages/netsuite-adapter/src/client/client.ts

```diff
@@ -206,50 +249,88 @@ export default class NetsuiteClient {
       .toArray()
   }
 
-  public static async createDependencyMap(
+  private static async getElemIdsAndCustInfos(
     changes: ReadonlyArray<Change>,
     deployReferencedElements: boolean,
     elementsSourceIndex: LazyElementsSourceIndexes,
-  ):Promise<Map<string, Set<string>>> {
-    const dependencyMap = new DefaultMap<string, Set<string>>(() => new Set())
-    const elements = changes.map(getChangeData)
-    const elemIdSet = new Set(elements.map(element => element.elemID.getFullName()))
-    const elemIdsAndCustInfoArr = await awu(elements)
-      .filter(element => !isField(element) || !elemIdSet.has(element.parent.elemID.getFullName()))
-      .map(async element => ({
-        elemId: element.elemID,
-        custInfos: await NetsuiteClient.toCustomizationInfos(
-          [element], deployReferencedElements, elementsSourceIndex
+  ): Promise<SDFObjectNode[]> {
+    const elementsAndChangeTypes = changes
+      .filter(isAdditionOrModificationChange)
+      .map(change =>
+        ({ element: getChangeData(change), changeType: getChangeType(change) }))
+    const elemIdSet = new Set(elementsAndChangeTypes.map(elementsAndChangeType =>
+      elementsAndChangeType.element.elemID.getFullName()))
+    return awu(elementsAndChangeTypes)
+      .filter(elementsAndChangeType => !isField(elementsAndChangeType.element)
+      || !elemIdSet.has(elementsAndChangeType.element.parent.elemID.getFullName()))
+      .map(async elementsAndChangeType => ({
+        elemIdFullName: elementsAndChangeType.element.elemID.getFullName(),
+        scriptid: getScriptIdFromElement(elementsAndChangeType.element),
+        changeType: elementsAndChangeType.changeType,
+        customizationInfos: await NetsuiteClient.toCustomizationInfos(
+          [elementsAndChangeType.element], deployReferencedElements, elementsSourceIndex
         ),
-      })).toArray()
+      }))
+      .toArray()
+  }
 
-    elemIdsAndCustInfoArr.forEach(elemIdAndCustInfo => {
-      elemIdAndCustInfo.custInfos.forEach(custInfo =>
+  public static async createDependencyMapAndGraph(
+    changes: ReadonlyArray<Change>,
+    deployReferencedElements: boolean,
+    elementsSourceIndex: LazyElementsSourceIndexes,
+  ):Promise<DependencyInfo> {
+    const dependencyMap = new DefaultMap<string, Set<string>>(() => new Set())
+    const elemIdsAndCustInfos = await NetsuiteClient.getElemIdsAndCustInfos(
+      changes, deployReferencedElements, elementsSourceIndex
+    )
+    const dependencyGraph = new Graph<SDFObjectNode>(
+      'elemIdFullName', elemIdsAndCustInfos.map(elemIdsAndCustInfo => new GraphNode(elemIdsAndCustInfo))
```

Comment:
nit - create the graph nodes inside `Graph` constructor.
```suggestion
      'elemIdFullName', elemIdsAndCustInfos
```